RSSReader/src/main/java/com/polarnick/day06/NewFeedChecker.java

```
@@ -0,0 +1,103 @@
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import java.io.IOException;
+import java.io.StringReader;
+
+/**
+ * Date: 23.10.13
+ *
+ * @author Nickolay Polyarniy aka PolarNick
+ */
+public class NewFeedChecker extends IntentService {
+    public static final String FEED_KEY = "feed";
+    public static final String RECEIVER_KEY = "receiver";
+
+    private static final int PERIOD_OF_REFRESH = 5 * 60 * 1000;//5 minutes
+
+    public NewFeedChecker() {
+        this("NewFeedCheckerIntentService");
+    }
+
+    public NewFeedChecker(String name) {
+        super(name);
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        ResultReceiver receiver = intent.getParcelableExtra(RECEIVER_KEY);
+        Feed lastDownloadedFeed = (Feed) intent.getSerializableExtra(FEED_KEY);
+
+        Feed actualFeed = null;
+        String url = lastDownloadedFeed.getUrl();
+        try {
+            ConnectivityManager connManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+            NetworkInfo mWifi = connManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
+
+            if (mWifi.isConnected()) {
+                SAXParserFactory factory = SAXParserFactory.newInstance();
+                SAXParser parser = factory.newSAXParser();
+
+                HttpResponse httpResponse = new DefaultHttpClient().execute(new HttpGet(url));
+                HttpEntity httpEntity = httpResponse.getEntity();
+
+                String xml = EntityUtils.toString(httpEntity, "UTF-8");
+                InputSource is = new InputSource(new StringReader(xml));
+                RSSHandler handler = new RSSHandler();
+                parser.parse(is, handler);
+
+                actualFeed = handler.retrieveFeed();
+                actualFeed.setUrl(url);
+            }
+        } catch (IOException e) {
+        } catch (ParserConfigurationException e) {
+        } catch (SAXException e) {
```

you should not ignore exception